cloudinit/sources/__init__.py

```diff
@@ -321,7 +321,7 @@ def override_ds_detect(self) -> bool:
         does not run, _something_ needs to detect the kernel command line
         definition.
         """
-        if self.dsname == parse_cmdline():
+        if self.dsname.lower() == parse_cmdline():
```

Comment:
+1 on standardizing on lower(), but for backward compat, I think we may want to support Case insensitive or at least the slightly more strict `parse_cmdline() in (self.dsname, self.dsname.lower())`